### ISSUE TITLE
Nested interactive controls are not announced by screen readers

### ISSUE THREAD

**Issue by AhmedAbdoOrtiga** [dequelabs/axe-core#3202]
Product: <<axe-core | axe Extension | axe Linter>>
axe-core

Expectation: << Describe what you expect the product to do >>
Focusable elements with an interactive control ancestor (any element that accepts user input such as button or anchor elements) are **not announced** by screen readers and create an empty tab stop. 

Actual: << Describe what the product actually does >>
Focusable elements with an interactive control ancestor (any element that accepts user input such as button or anchor elements) are **announced** by screen readers (NVDA, Narrator, etc..). 

Motivation: << Describe why you want the behavior to be changed >>
Screen readers are able to announce correctly nested interactive elements so we were wondering if we can  
<pre><code>
axe-core version: 4.3.2
axe-webdriver, extension or other integration version: X.Y.Z

Browser and Assistive Technology versions

For Tooling issues:
- Node version: XX   Any
- Platform:   Any
</code></pre>
